Content.Server/Atmos/GasSprayerComponent.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.Interfaces;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Pointing;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Server.Interfaces.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Maths;
+using Robust.Shared.Serialization;
+
+
+namespace Content.Server.Atmos
+{
+    [RegisterComponent]
+    public class GasSprayerComponent : Component, IAfterInteract
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager = default!;
+        [Dependency] private readonly IServerEntityManager _serverEntityManager = default!;
+#pragma warning restore 649
+
+        //TODO: create a function that can create a gas based on a solution mix
+        public override string Name => "GasSprayer";
+
+        private string _spraySound;
+        private string _sprayType;
+        private string _fuelType;
+        private string _fuelName;
+        private int _fuelCost;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _spraySound, "spraySound", string.Empty);
+            serializer.DataField(ref _sprayType, "sprayType", string.Empty);
+            serializer.DataField(ref _fuelType, "fuelType", string.Empty);
+            serializer.DataField(ref _fuelName, "fuelName", "fuel");
+            serializer.DataField(ref _fuelCost, "fuelCost", 50);
+        }
+
+        public void AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            if (Owner.TryGetComponent(out SolutionComponent tank) &&
+                tank.Solution.GetReagentQuantity(_fuelType) == 0)
+            {
```

Comment:
If SolutionComponent was ensured in this component's initialize method it would make more sense to potentially ignore this, however as it stands right now you could add this component to an entity without a solution component which would lead to an error.

Even if it was ensured however, it still wouldn't make sense to have a branch of code where tank is potentially null, as components can change at any time.

```suggestion
            if (!Owner.TryGetComponent(out SolutionComponent tank))
                return;

            if (tank.Solution.GetReagentQuantity(_fuelType) == 0)
            {
```

Reply:
I suppose it would make sense if they somehow got decoupled.